posthog/api/test/test_trends.py

```diff
@@ -112,30 +116,106 @@ def test_includes_only_intervals_within_range(client: Client):
+    project = get_current_project_ok(client=client)
+
+    # "2021-09-19" is a sunday, i.e. beginning of week
+    with freeze_time("2021-09-20T16:00:00"):
+        #  First identify as a member of the cohort
+        distinct_id = "abc"
+        identify(distinct_id=distinct_id, team_id=project["id"])
+
+        capture_event(
+            event=EventData(
+                event="instance status report",
+                team_id=project["id"],
+                distinct_id=distinct_id,
+                timestamp=datetime(2021, 9, 3),
+                properties={"distinct_id": "abc", "teams__1__events_count_by_lib__posthog-python": "is_set"},
+            )
+        )
+
+        event = {
+            "id": "instance status report",
+            "math": "dau",
+            "name": "instance status report",
+            "type": "events",
+            "order": 0,
+            "properties": [],
+            "math_property": None,
+        }
+
+        trends = get_trends_ok(
+            client,
+            request=TrendsRequest(
+                date_from="2021-09-3",
+                date_to="2021-09-3",
+                interval="day",
+                insight="TRENDS",
+                display="ActionsTable",
+                funnel_window_days=14,
+                events=json.dumps([event]),
+            ),
+        )
+
+        assert trends == {
+            "is_cached": False,
+            "last_refresh": "2021-09-20T16:00:00Z",
+            "next": None,
+            "result": [
+                {
+                    "action": event,
+                    "label": "instance status report",
+                    "aggregated_value": 1,
+                    "count": 1.0,
+                    "data": [1.0],
+                    # Prior to the fix this would also include '29-Aug-2021'
+                    "labels": ["3-Sep-2021"],
+                    "days": ["2021-09-03"],
+                }
+            ],
+        }
+
+
 @dataclasses.dataclass
 class TrendsRequest:
     date_from: str
     date_to: str
     interval: str
     insight: str
-    breakdown: List[int]
-    breakdown_type: str
     display: str
-    events: List[Dict[str, Any]]
+    # Events needs to be encoede as a json string
+    events: str
+    breakdown: Optional[List[int]] = None
+    breakdown_type: Optional[str] = None
+    funnel_window_days: Optional[int] = None
 
 
 def get_trends(client, request: TrendsRequest):
     return client.get(
         "/api/insight/trend/",
         data={
-            "date_from": request.date_from,
-            "date_to": request.date_to,
-            "interval": request.interval,
-            "insight": request.insight,
-            "breakdown": request.breakdown,
-            "breakdown_type": request.breakdown_type,
-            "display": request.display,
-            "events": json.dumps(request.events),
+            key: value
+            for key, value in dataclasses.asdict(request).items()
```

Comment:
Does this work? events was json dumped before - and it needs to be to actually work.

Reply:
"events" in the `TrendsRequest` is a string 🤞